File: dbxml/src/java/com/sleepycat/dbxml/XmlUpdateContext.java
//
// See the file LICENSE for redistribution information.
//
// Copyright (c) 2002,2009 Oracle.  All rights reserved.
//
//

package com.sleepycat.dbxml;
/* Phasing this class out.  The XmlManager is stored so that a new
 * XmlUpdateContext object can be created on the C++ side. 
 */
public class XmlUpdateContext {
    protected XmlManager manager;
	
    protected static long getCPtr(XmlUpdateContext obj) {
	if(obj == null)
	    obj.toString();  //force a null pointer exception
	return XmlManager.getCPtr(obj.manager);
    }
	
    protected XmlUpdateContext(XmlManager mgr){
	manager = mgr;
    }
}

File: dbxml/src/dbxml/XmlUpdateContext.cpp
//
// See the file LICENSE for redistribution information.
//
// Copyright (c) 2002,2009 Oracle.  All rights reserved.
//
//

#include "DbXmlInternal.hpp"
#include "dbxml/XmlUpdateContext.hpp"
#include "dbxml/XmlContainer.hpp"
#include "UpdateContext.hpp"
#include "Container.hpp"

using namespace DbXml;

XmlUpdateContext::XmlUpdateContext()
	: updateContext_(0)
{
}

XmlUpdateContext::XmlUpdateContext(UpdateContext *uc)
	: updateContext_(uc)
{
	if (updateContext_)
		updateContext_->acquire();
}

XmlUpdateContext::~XmlUpdateContext()
{
	if (updateContext_)
		updateContext_->release();
}

XmlUpdateContext::XmlUpdateContext(const XmlUpdateContext &o)
	: updateContext_(o.updateContext_)
{
	if (updateContext_ != 0)
		updateContext_->acquire();
}

XmlUpdateContext &XmlUpdateContext::operator=(const XmlUpdateContext &o)
{
	if (this != &o && updateContext_ != o.updateContext_) {
		if (updateContext_ != 0)
			updateContext_->release();
		updateContext_ = o.updateContext_;
		if (updateContext_ != 0)
			updateContext_->acquire();
	}
	return *this;
}
